### ISSUE TITLE
Winget updated my Mozilla Firefox and instead of downloading the Portuguese version, it downloaded the English version

### ISSUE THREAD

**Issue by maisondasilva** [microsoft/winget-cli#3755]
### Brief description of your issue

Winget updated my Mozilla Firefox and instead of downloading the Portuguese version, it downloaded the English version

### Steps to reproduce

![image](https://github.com/microsoft/winget-cli/assets/11423362/dffa2344-a915-4c42-8453-df71cff7e564)

### Expected behavior

Mozilla Firefox Pt-br

### Actual behavior

Mozilla Firefox En-us

### Environment

```shell
v1.7.2782-preview
```

**Comment by denelon** [comment]
It's a workaround, but you can pass "--locale" to get a specific locale when upgrading a single package.

We don't have a reliable way to parse the locale from the "Name"/"displayName".

**Comment by maisondasilva** [comment]
> It's a workaround, but you can pass "--locale" to get a specific locale when upgrading a single package.
> 
> We don't have a reliable way to parse the locale from the "Name"/"displayName".

Thanks for the update, the command would look like this right?

`winget upgrade --all --locale pt-BR`

**Comment by denelon** [comment]
That might not have the desired affect because not all packages have that locale so it might not install upgrades for things that don't offer the specified locale.

Maybe running that followed by running it without the locale would update the rest.

I was suggesting `winget upgrade Mozilla.Firefox --locale pt-BR`

**Comment by maisondasilva** [comment]
@denelon Would it be a solution for me to run the command `winget upgrade --all --locale pt-BR` and then `winget upgrade --all`?
Because for example Winrar updates with the Brazilian version, the problem is that I know which program will update with the language correctly and which ones won't.

Thanks for the time!

Thanks Maison

**Comment by denelon** [comment]
I don't remember if the CLI argument is a "prefer" logic or a "require" logic. I'll ask @JohnMcPMS...

**Comment by maisondasilva** [comment]
@denelon I tried updating using  `winget upgrade --all --locale pt-BR` ,but it doesn't install if I use winget upgrade --all it will install the English version again

Is there any way to fix this and have winget install the version in my language?

![image](https://github.com/microsoft/winget-cli/assets/11423362/330daa1c-42ec-4031-be44-0d99cfd0dd51)

**Comment by dftf-stu** [comment]
@maisondasilva : it looks like in the years since your ticket, each language for Firefox is now a separate entry

So `Mozilla.Firefox.pt-BR` is now what you're after (`Mozilla.Firefox` alone always refers to the en-US version).

Not sure if you'd need to remove your current install, or if `winget upgrade Mozilla.Firefox.pt-BR` would be enough?

**Comment by maisondasilva** [comment]
Yes, that sounds better, but when there's an update, will it recognize the Brazilian Portuguese version?

<img width="765" height="341" alt="Image" src="https://github.com/user-attachments/assets/ebf05b99-fdaa-471c-88d4-2416b61264e4" />

**Comment by dftf-stu** [comment]
@maisondasilva : I'd guess it's best to uninstall your current Firefox, then run `winget install Mozilla.Firefox.pt-BR`
Then that package will keep being updated (for as long as someone volunteers to add each new update into it, of course)

It should link back to your current profile-data (bookmarks, history, passwords), as if you ever install Firefox manually (outside of winget), then uninstall, and install a different language, it still looks for profiles in the same area.  The only time it wouldn't is if you were to install Firefox Beta or Firefox Nightly (or run a portable version), as the profiles are separate between them.

If you're concerned though, do one of these things first:
- Create a System Restore point
- If your Firefox is the latest version, go to Menu (the three lines) > ⚙Settings > Sync and use the new "Backup" option there to create a backup of your profile to a local folder
- Or backup your current Firefox profile data manually via these commands:
robocopy "%localappdata%\Mozilla\Firefox" "E:\FirefoxBackup\Local" /COPY:DATX /DCOPY:DAT /XJ
robocopy "%appdata%\Mozilla\Firefox" "E:\FirefoxBackup\Roaming" /COPY:DATX /DCOPY:DAT /XJ
- You can change "E:\FirefoxBackup" to any local folder (e.g. "C:\BACKUP\FirefoxBackup"), or the letter E to a USB drive inserted
